torchgeo/datasets/spacenet.py

```diff
@@ -737,6 +738,242 @@ def _load_files(self, root: str) -> List[Dict[str, str]]:
+
+    .. versionadded:: 0.2
+    """
+
+    dataset_id = "spacenet5"
+    collection_md5_dict = {
+        "sn5_AOI_7_Moscow": "b18107f878152fe7e75444373c320cba",
+        "sn5_AOI_8_Mumbai": "1f1e2b3c26fbd15bfbcdbb6b02ae051c",
+    }
+
+    imagery = {
+        "MS": "MS.tif",
+        "PAN": "PAN.tif",
+        "PS-MS": "PS-MS.tif",
+        "PS-RGB": "PS-RGB.tif",
+    }
+    chip_size = {
+        "MS": (325, 325),
+        "PAN": (1300, 1300),
+        "PS-MS": (1300, 1300),
+        "PS-RGB": (1300, 1300),
+    }
+    label_glob = "labels.geojson"
+
+    def __init__(
+        self,
+        root: str,
+        image: str = "PS-RGB",
+        speed_mask: Optional[bool] = False,
+        collections: List[str] = [],
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        download: bool = False,
+        api_key: Optional[str] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new SpaceNet 5 Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            image: image selection which must be in ["MS", "PAN", "PS-MS", "PS-RGB"]
+            speed_mask: use multi-class speed mask (created by binning roads at
+                10 mph increments) as label if true, else use binary mask
+            collections: collection selection which must be a subset of:
+                         [sn5_AOI_7_Moscow, sn5_AOI_8_Mumbai]
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory.
+            api_key: a RadiantEarth MLHub API key to use for downloading the dataset
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing
+        """
+        assert image in {"MS", "PAN", "PS-MS", "PS-RGB"}
+        self.speed_mask = speed_mask
+        super().__init__(
+            root, image, collections, transforms, download, api_key, checksum
+        )
+
+    def _load_mask(
+        self, path: str, tfm: Affine, raster_crs: CRS, shape: Tuple[int, int]
+    ) -> Tensor:
+        """Rasterizes the dataset's labels (in geojson format).
+
+        Args:
+            path: path to the label
+            tfm: transform of corresponding image
+            shape: shape of corresponding image
+
+        Returns:
+            Tensor: label tensor
+        """
+        min_speed_bin = 1
+        max_speed_bin = 65
+        speed_arr_bin = np.arange(min_speed_bin, max_speed_bin + 1)
+        bin_size_mph = 10.0
+        speed_cls_arr = np.array(
+            [int(math.ceil(s / bin_size_mph)) for s in speed_arr_bin]
+        )
+
+        try:
+            with fiona.open(path) as src:
+                vector_crs = CRS(src.crs)
```

Comment:
Isn't `src.crs` already a `CRS` object?

Reply:
With fiona, it's epsg dict i.e. `{'init': 'epsg:4326'}`